proteuslib/tests/test_user_scaling_with_chlorination.py

```diff
@@ -0,0 +1,838 @@
+
+        for i in model.fs.unit.control_volume.equilibrium_reaction_extent_index:
+            scale = value(model.fs.unit.control_volume.reactions[0.0].k_eq[i[1]].expr)
+            iscale.set_scaling_factor(model.fs.unit.control_volume.equilibrium_reaction_extent[0.0,i[1]], 10/scale)
+            iscale.constraint_scaling_transform(model.fs.unit.control_volume.reactions[0.0].
+                    equilibrium_constraint[i[1]], 0.1)
+
+        # Next, try adding scaling for species
+        min = 1e-6
+        for i in model.fs.unit.control_volume.properties_out[0.0].mole_frac_phase_comp:
+            # i[0] = phase, i[1] = species
+            if model.fs.unit.inlet.mole_frac_comp[0, i[1]].value > min:
+                scale = model.fs.unit.inlet.mole_frac_comp[0, i[1]].value
+            else:
+                scale = min
+            iscale.set_scaling_factor(model.fs.unit.control_volume.properties_out[0.0].mole_frac_comp[i[1]], 10/scale)
+            iscale.set_scaling_factor(model.fs.unit.control_volume.properties_out[0.0].mole_frac_phase_comp[i], 10/scale)
+            iscale.set_scaling_factor(model.fs.unit.control_volume.properties_out[0.0].flow_mol_phase_comp[i], 10/scale)
+            iscale.constraint_scaling_transform(
+                model.fs.unit.control_volume.properties_out[0.0].component_flow_balances[i[1]], 10/scale)
+            iscale.constraint_scaling_transform(model.fs.unit.control_volume.material_balances[0.0,i[1]], 10/scale)
+
+        iscale.calculate_scaling_factors(model.fs.unit)
+
+        assert isinstance(model.fs.unit.control_volume.scaling_factor, Suffix)
+
+        assert isinstance(model.fs.unit.control_volume.properties_out[0.0].scaling_factor, Suffix)
+
+        assert isinstance(model.fs.unit.control_volume.properties_in[0.0].scaling_factor, Suffix)
+
+        assert isinstance(model.fs.unit.control_volume.reactions[0.0].scaling_factor, Suffix)
+
+        # Before calling the 'constraint_autoscale_large_jac', we need to
+        #   set some 'state_args' for all system variables
+
+        # NOTE: Give an initial guess of pH 7 and inlet values or 1e-6 for others
+        state_args = {'mole_frac_comp':
+                        {   'H2O': 1,
+                            'HOCl': 4e-6,
+                            'H_+': 10**-7/55.6,
+                            'NCl3': 1e-6,
+                            'NH2Cl': 1e-6,
+                            'NH3': 1e-6,
+                            'NH4_+': 1e-6,
+                            'NHCl2': 1e-6,
+                            'OCl_-': 1e-6,
+                            'OH_-': 10**-7/55.6
+                        },
+                        'pressure': 101325,
+                        'temperature': 300,
+                        'flow_mol': 10
+                    }
+        flags = fix_state_vars(model.fs.unit.control_volume.properties_out, state_args)
+        revert_state_vars(model.fs.unit.control_volume.properties_out, flags)
```

Comment:
This is setting up initial values for variables of the model using 'state_args'. The values given in the state args are just the inlet values, or an arbitrary small (but not too small) value for ions, and an assumed pH of 7 to set H_+ and OH_-.